cli/src/lib.rs

```diff
@@ -3789,8 +3790,7 @@ fn create_idl_buffer(
     idl: &Idl,
     priority_fee: Option<u64>,
 ) -> Result<Pubkey> {
-    let keypair = solana_sdk::signature::read_keypair_file(keypair_path)
-        .map_err(|_| anyhow!("Unable to read keypair file"))?;
+    let keypair = get_keypair(cfg.provider.wallet.to_string())?;
```

Comment:
Same here, this should use `keypair_path` instead:

```suggestion
    let keypair = get_keypair(keypair_path)?;
```

Reply:
Same here.